warehouse/forklift/legacy.py

```diff
@@ -394,6 +394,37 @@ def _validate_classifiers(form, field):
             )
 
 
+def _validate_dynamic(_form, field):
+    declared_dynamic_fields = {str.title(k) for k in field.data or []}
+    disallowed_dynamic_fields = {"Name", "Version", "Metadata-Version"}
+    if invalid := (declared_dynamic_fields & disallowed_dynamic_fields):
+        raise wtforms.validators.ValidationError(
+            f"The following metadata field(s) cannot be marked as dynamic: {invalid!r}",
+        )
+    allowed_dynamic_fields = (
+        {field.description for field in MetadataForm() if field.description}
+    ) - disallowed_dynamic_fields
+    if invalid := (declared_dynamic_fields - allowed_dynamic_fields):
+        raise wtforms.validators.ValidationError(
+            f"The following values are not valid for Dynamic: {invalid!r}"
+        )
```

Comment:
```suggestion
    declared_dynamic_fields = {str.title(k) for k in field.data or []}
    disallowed_dynamic_fields = {"Name", "Version", "Metadata-Version"}
    if invalid := (declared_dynamic_fields & disallowed_dynamic_fields):
        raise wtforms.validators.ValidationError(
            f"The following metadata field(s) are valid, but cannot be marked as dynamic: {invalid!r}",
        )
    allowed_dynamic_fields = set(DynamicFieldsEnum.enums)
    if invalid := (declared_dynamic_fields - allowed_dynamic_fields):
        raise wtforms.validators.ValidationError(
            f"The following metadata field(s) are not valid and cannot be marked as dynamic: {invalid!r}"
        )
```
I think it's better to use an explicit Enum for this, because it's not actually guaranteed that `description` is set for all these fields. Also, ideally, we would reuse that same enum in the migration as well.

Reply:
If we import the enum in the migration, wouldn't this break the migration if we ever update the enum? (Sorry if that's a stupid question, i don't have any experience with alembic)

Reply:
It's a good question. It wouldn't necessarily break the migration itself (because it would always pull the latest enum when we were running the migration), but it might make the migration technically incorrect if that were ever to happen. I'm OK with not using the enum -- we can verify that they are the same before merging.

Reply:
With that resolved, is there anything else you need to be changed to be able to merge #15452?

Reply:
No, I think its fine!